src/samplers/lumigoSampler.test.js

```diff
@@ -1,217 +1,145 @@
-import { extractUrl, shouldSkipSpanOnRouteMatch } from './lumigoSampler';
+import { doesEndpointMatchRegexes, extractEndpoint, parseStringToArray } from './lumigoSampler';
+import { SpanKind } from '@opentelemetry/api';
 
 describe('lumigo sampler', () => {
   afterEach(() => {
     jest.clearAllMocks();
   });
 
   [
-    // Test happy flow - regex matches url
     {
-      url: 'https://example.com',
-      regex: '.*example.*',
-      shouldSkip: true,
+      rawArrayString: '["a"]',
+      expectedArray: ['a'],
     },
     {
-      url: 'https://example.com',
-      regex: 'https://example.com',
-      shouldSkip: true,
+      rawArrayString: '["a", "b"]',
+      expectedArray: ['a', 'b'],
     },
     {
-      url: 'https://example.com/about',
-      regex: 'https://example.com/about',
-      shouldSkip: true,
+      rawArrayString: '[]',
+      expectedArray: [],
     },
     {
-      url: 'https://example.com/about',
-      regex: 'https?://.+/about',
-      shouldSkip: true,
+      rawArrayString: '"a","b"',
+      expectedArray: [],
     },
-
-    // Test url doesn't match regex
     {
-      url: 'https://example.com',
-      regex: '.*not-example.*',
-      shouldSkip: false,
+      rawArrayString: 'Not an array',
+      expectedArray: [],
     },
     {
-      url: 'https://example.com',
-      regex: 'https://not-example.com',
-      shouldSkip: false,
+      rawArrayString: null,
+      expectedArray: [],
     },
     {
-      url: 'https://example.com/about',
-      regex: 'https://example.com/not-about',
-      shouldSkip: false,
+      rawArrayString: undefined,
+      expectedArray: [],
+    },
+    {
+      rawArrayString: '["a", 2]',
+      expectedArray: [],
     },
-
-    // Test regex is invalid
     {
-      url: 'https://example.com',
-      // The regex is invalid because a char must come before the * operator
-      regex: '*',
-      shouldSkip: false,
+      rawArrayString: '["a", true]',
+      expectedArray: [],
     },
+  ].map(({ rawArrayString, expectedArray }) => {
```

Comment:
The jest way is using `test.each`: https://jestjs.io/docs/api#2-testeachtablename-fn-timeout if you're up for it, it reduces verbosity greatly

Reply:
Thanks, implemented it that way